File: java-course/src/test/java/com/example/selenium/spree/HomePage.java
package com.example.selenium.spree;

import org.openqa.selenium.remote.RemoteWebDriver;
import org.testng.Assert;

public class HomePage {
    public static final String URL = "https://spreecommerce-demo.herokuapp.com/";
    public static final String TITLE = "Spree Demo Site";

    private RemoteWebDriver webDriver;

    public HomePage(RemoteWebDriver webDriver) {
        this.webDriver = webDriver;
    }

    public void validateTitle() {
        String title = webDriver.getTitle();
        Assert.assertEquals(title, TITLE);
    }

    public void validateUrl() {
        String url = webDriver.getCurrentUrl();
        Assert.assertEquals(url, URL);
    }
}

File: python-course/com/example/selenium/spree/home_page.py
from com.example.selenium.spree.spree_page import SpreePage, url_to_be


class HomePage(SpreePage):

    def __init__(self, test):
        # self.test = test
        super().__init__(test, url_to_be("https://selenium.jacobparr.com/"))

    def validateTitle(self):
        title = self.test.webDriver.title
        self.test.assertEquals("Spree Demo Site", title)

    def validateUrl(self):
        url = self.test.webDriver.current_url
        self.test.assertEquals("https://selenium.jacobparr.com/", url)